evm_arithmetization/src/prover.rs

```diff
@@ -424,15 +433,30 @@ pub mod testing {
     pub fn simulate_execution_all_segments<F>(
         inputs: GenerationInputs<F>,
         max_cpu_len_log: usize,
+        struct_logs: Option<Vec<Option<Vec<ZeroStructLog>>>>,
     ) -> Result<(), ErrorWithTries<SegmentError>>
     where
         F: RichField,
     {
         features_check(&inputs.clone().trim());
 
-        for segment in SegmentDataIterator::<F>::new(&inputs, Some(max_cpu_len_log)) {
-            segment?;
-        }
+        match struct_logs {
+            Some(struct_log) => {
+                for segment in SegmentDataIterator::<F>::new(
+                    &inputs,
+                    Some(max_cpu_len_log),
+                    &Some(&struct_log.to_owned()),
+                ) {
+                    segment?;
+                }
+            }
+            None => {
+                for segment in SegmentDataIterator::<F>::new(&inputs, Some(max_cpu_len_log), &None)
+                {
+                    segment?;
+                }
+            }
+        };
```

Comment:
I would probably pass ownership of the `struct_logs` to the `SegmentDataIterator::new` and then this functions becomes:

```
        for segment in SegmentDataIterator::<F>::new(
            &inputs,
            Some(max_cpu_len_log),
            struct_logs,
        ) {
            segment?;
        }
```